lib/chef/resource/windows_certificate.rb

```diff
@@ -261,9 +256,11 @@ def get_thumbprint(store_name, location, source)
         end
 
         def resolve_thumbprint(thumbprint)
-          return thumbprint if valid_thumbprint?(thumbprint)
-
-          powershell_exec!(get_thumbprint(new_resource.store_name, ps_cert_location, new_resource.source)).result
+          # valid_thumbprint can return false under at least 2 conditions:
+          # one is that the thumbprint is in fact busted
+          # the second is that the thumbprint is valid but belongs to an expired certificate already installed
```

Comment:
This comment does not seem accurate.  Above, `valid_thumbprint?` is defined as only: 
```
 string.match?(/[0-9A-Fa-f]/) && string.length == 40
```
`resolve_thumbprint` in its current form only validates the correctness of the thumbprint.  Is that the  intent?

Reply:
I updated that method name to validate_thumbprint.